internal/policy/policy.go

```diff
@@ -193,6 +198,32 @@ func LoadStateForEntry(ctx context.Context, repo *git.Repository, e rsl.EntryTyp
 	return state, nil
 }
 
+func (s *State) FindAuthorizedSigningKeyIDs(ctx context.Context, roleName string) ([]string, error) {
+	if err := s.Verify(ctx); err != nil {
+		return []string{}, err
+	}
+
+	rootMetadata, err := s.GetRootMetadata()
+	if err != nil {
+		return []string{}, err
+	}
+
+	if roleName == RootRoleName {
+		return rootMetadata.Roles[RootRoleName].KeyIDs, nil
+	}
+
+	if roleName == TargetsRoleName {
+		return rootMetadata.Roles[TargetsRoleName].KeyIDs, nil
```

Comment:
If we want to be safe, we should double check this key returned non-nil before we try to access the KeyIDs field.